### ISSUE TITLE
Impossible the see usage of a shared package

### ISSUE THREAD

**Issue by mhaack** [apache/incubator-openwhisk#3241]
At the moment it is not possible to see the usage of aka. bindings to shared package.

Use case:
* In the provider namespace create a shared package and add some actions to be re-used
* In the consumer namespace bind the shared package and use the actions in a sequence 

As a provider of actions via a shared package it would be nice to have a "see who consumes my package" feature as part of the OW API and cli.

**Comment by rabbah** [comment]
What kind of information would you want to back link?

**Comment by mhaack** [comment]
I think that's a list of namespaces that have bindings to my public packages. The intention is to get at least a list of namespaces using my public packages, for example if I plan to update the actions in a public package they might be affected by the changes I make.

For example:
```
⇒  wsk package list-bindings
packages
/my-namespace/package1                           used by namespace /other-namespace-a
/my-namespace/package2                           used by namespace /other-namespace-b
```

This could be a the minimum information without exposing to much details about the namespaces consuming my shared packages.

**Comment by rabbah** [comment]
For this kind of thing why not use versioning; it would be a better approach. You could add a version to the package name (version at user level) and follow semver protocol.

Since a package is entirely public someone who wants to protect against a breaking change without versioning would/could make a copy of the code. It isn’t possible to know who made a copy.

**Comment by mhaack** [comment]
I agree on the package versioning, that is and idea we are currently looking into. Copying is out of scope, if some copies an action he is on his own.

For example can I delete an action in a shared package or the entire package? For such a case it would be nice if I can see who uses "my" public shared package.

**Comment by rabbah** [comment]
If the goal is to prevent breaking changes, then versioning is the preferred route. 
We could provide a count of references that exist - but providing the namespaces could be a reason for concern. Also note that actions in a public package may be called directly without a binding.

**Comment by rabbah** [comment]
A related thought - today when you invoke an action through a binding, the resolution to the package happens fairly early on. The activation record that's created does not note that the activation came through a binding.

The activation count for an action in a package is owned by the caller. It may be useful to provide metrics that show the owner of a package how often their actions are invokes by others (through binding or directly).

I can that being useful - and for a catalog of packages/actions that are public, could serve as a way of providing meaningful information; eventually coupled with "stars" and recommendations.

**Comment by mhaack** [comment]
+1 for the activation count metrics for catalog of packages/actions that are public

**Comment by rabbah** [comment]
@upgle your PR https://github.com/apache/incubator-openwhisk/pull/4211 is related to my last comment above.
